### DISCUSSION TITLE
apolloGateway cannot connect to graphql express service

### DISCUSSION THREAD

**Discussion by Furkan-Gulsen** [apollographql/apollo-server#6693] [Q&A] [answered]
I ran a service using apolloGateway:

```javascript
export const server = async () => {
  const app = express();
  const services = [ { name: "product-service", url: "http://localhost:4001/graphql" }];
  const availableServices: any[] = [];

  const gateway = new ApolloGateway({
    serviceList: services,
    buildService: ({ name, url }) => {
      return new RemoteGraphQLDataSource({
        url,
      });
    },
  });

  const server = new ApolloServer({
    subscriptions: false,
    gateway,
  });

  await server.start();
  server.applyMiddleware({ app });

  return app.listen(3000);
};
```

Then I ran a service called product using Graphql-express:

```javascript
export const server = async () => {
  const schema = await buildSchema({
    resolvers: [__dirname + "/api/**/*.resolver.ts"],
    emitSchemaFile: path.resolve(__dirname, `../schema.gql`),
    authChecker: authChecker,
  });

  const app = express();

  const server = new ApolloServer({
    schema,
    context: (ctx: Context) => {
      const context = ctx;
      return context;
    },
  });

  await server.start();

  server.applyMiddleware({
    app,
  });

  return app.listen(4001);
};
```

Here I am getting an error on the api-gateway side to connect:

_Error checking for changes to service definitions: Couldn't load service definitions for "product-service" at http://localhost:4001/graphql: 400: Bad Request_

**Comment by glasser** [comment, marked as answer]
I suspect you're more likely to find help if you provide your reproduction as fully self-contained, eg a git repo that can be cloned or a codesandbox.io sandbox. Otherwise a lot of guesswork goes into the software versions you've chosen and it can be hard to nail down the real problem. This is also more appropriate for https://github.com/apollographql/federation as all the relevant logic is maintained in that project.

**Reply by glasser** [reply]
I suspect that you are not using `@apollo/subgraph` and its `buildSubgraphSchema` function though, which is what enables your subgraph to do things like "expose a `_service` field that shows its schema to the serviceList variety of gateway".
